src/gui/guiFormSpecMenu.cpp

```diff
@@ -1624,23 +1624,57 @@ void GUIFormSpecMenu::parseListColors(parserData* data, const std::string &eleme
 void GUIFormSpecMenu::parseTooltip(parserData* data, const std::string &element)
 {
 	std::vector<std::string> parts = split(element,';');
-	if (parts.size() == 2) {
-		std::string name = parts[0];
-		m_tooltips[name] = TooltipSpec(utf8_to_wide(unescape_string(parts[1])),
-			m_default_tooltip_bgcolor, m_default_tooltip_color);
+	if (parts.size() < 2) {
+		errorstream << "Invalid tooltip element(" << parts.size() << "): '"
+				<< element << "'"  << std::endl;
 		return;
 	}
 
-	if (parts.size() == 4) {
-		std::string name = parts[0];
-		video::SColor tmp_color1, tmp_color2;
-		if ( parseColorString(parts[2], tmp_color1, false) && parseColorString(parts[3], tmp_color2, false) ) {
-			m_tooltips[name] = TooltipSpec(utf8_to_wide(unescape_string(parts[1])),
-				tmp_color1, tmp_color2);
-			return;
-		}
+	// Get mode and check size
+	bool rect_mode = parts[0].find(',') != std::string::npos;
+	size_t base_size = rect_mode ? 3 : 2;
+	if (parts.size() != base_size && parts.size() != base_size + 2) {
+		errorstream << "Invalid tooltip element(" << parts.size() << "): '"
+				<< element << "'"  << std::endl;
+		return;
+	}
+
+	// Read colors
+	video::SColor bgcolor = m_default_tooltip_bgcolor;
+	video::SColor color   = m_default_tooltip_color;
+	if (parts.size() == base_size + 2 &&
+			(!parseColorString(parts[base_size], bgcolor, false) ||
+				!parseColorString(parts[base_size + 1], color, false))) {
+		errorstream << "Invalid color in tooltip element(" << parts.size()
+				<< "): '" << element << "'"  << std::endl;
+		return;
+	}
+
+	// Make tooltip spec
+	std::string text = unescape_string(parts[rect_mode ? 2 : 1]);
+	TooltipSpec spec(utf8_to_wide(text), bgcolor, color);
+
+	// Add tooltip
+	if (rect_mode) {
+		std::vector<std::string> v_pos  = split(parts[0], ',');
+		std::vector<std::string> v_geom = split(parts[1], ',');
+
+		MY_CHECKPOS("tooltip",  0);
```

Comment:
The extra space could make LINT fail

Reply:
It doesn't, and the extra space lines up with the line below